proposals/exp-show-target.md

```diff
@@ -0,0 +1,214 @@
+# Add targets/stages argument to dvc exp show
+
+- Enhancement Proposal PR: (leave this empty)
+- Contributors: dberenbaum, (add your Github handle)
+
+# Summary
+
+The `dvc exp show` command shows a comparison of the results of different
+experiments along with the parameters and metrics values for each experiment.
+The command's output currently includes:
+* All values that are defined in `params.yaml`.
+* All values in other referenced parameters files in any DVC stage.
+* All values in any referenced metrics files in any DVC stage.
+This proposal allows users to see only parameters and metrics relevant to
+specified pipeline stages.
+
+See https://github.com/iterative/dvc/issues/5451 for more background.
+
+# Motivation
+
+`dvc exp show` is the main view by which users may compare experiments, and it's
+becoming useful for other purposes, like comparing multiple commits (see
```

Comment:
> it's becoming useful for other purposes, like comparing multiple commits

Maybe we should discuss that too/separately (as a proposal) 🙂 Should this be a top-level command? Perhaps `dvc compare (--exp[eriments])`

Although, `dvc compare` is also the name I proposed to unify the different `diff` subcommands at the top-level (another possibly interesting proposal doc).

Reply:
> discuss that too/separately (as a proposal)

Meta: how do we pick between / when to we move from https://github.com/iterative/dvc/discussions to https://github.com/iterative/enhancement-proposals ?

Reply:
> Meta: how do we pick between / when to we move from https://github.com/iterative/dvc/discussions to https://github.com/iterative/enhancement-proposals ?

Yes, I'm struggling with that already! Let's discuss at retro.

Reply:
This really belongs in a separate discussion, but it seems like we will eventually want to unify all of the `diff` commands.

`exp diff` exists as a convenience wrapper for params + metrics diff, but all of the individual `dvc ... diff` commands (including `dvc diff` itself) already support comparing any git commit or reference (including shortened DVC experiment names). It would probably make more sense to just have one `diff` command and where you can specify which output (data/metrics/params) you want to include/exclude (and nothing about this would be specific to experiments).

Reply:
Funny you say that: https://github.com/iterative/enhancement-proposals/tree/unified-show-diff 😉